test/integration/integration_test.cc

```diff
@@ -235,8 +235,8 @@ TEST_P(IntegrationTest, ResponseFramedByConnectionCloseWithReadLimits) {
 
   auto response = codec_client_->makeHeaderOnlyRequest(default_request_headers_);
   waitForNextUpstreamRequest();
-  // Disable chunk encoding to trigger framing by connection close.
-  upstream_request_->encodeHeaders(Http::TestHeaderMapImpl{{":status", "200"}, {":no-chunks", "1"}},
+  // Send illegal pseudo-header to trigger framing by connection close.
+  upstream_request_->encodeHeaders(Http::TestHeaderMapImpl{{":status", "200"}, {":illegal", "1"}},
```

Comment:
I'm pretty sure that disabling chunk encoding was not the reason for the connection close in the old code.

cc @antoniovicente

Reply:
I'm pretty sure ":no-chunks" is necessary to get the behavior I was looking for.
Perhaps I should have been more strict about the assertions on this test regarding the client getting a 200 response with 512 bytes in the response body.

Please undo this change.

Reply:
Change LGTM other than this comment thread.

/wait

Reply:
The test still passes with this change, even if I add:
```
  EXPECT_THAT(response->headers(), HttpStatusIs("200"));
  EXPECT_EQ(512, response->body().size());
```

I will undo the change, but it's unclear what is this really testing...

Reply:
Already fixed, PTAL.

Reply:
I'll take a closer look when I'm back from vacation.  Thanks for rolling back this change for now, this thread in the original review thread may provide some context: https://github.com/envoyproxy/envoy/pull/9509#discussion_r363533634